### PR TITLE
[Feature/PI-250] Make a request to cpm endpoint search

### PR DESCRIPTION
**Pull request by jameslinnell**
## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: docker/sds-api/spine-directory-service/sds/request/cpm.py [hunk 1]
```diff
@@ -38,6 +50,41 @@ def process_cpm_device_request(data: dict, query_parts: dict):
     filtered_devices = devices.filter_cpm_response()
     return devices.transform_to_ldap(filtered_devices)
 
+def make_get_request(call_name: str, url, headers=None, params=None):
+    res = requests.get(url, headers=headers, params=params)
+    handle_error(res, call_name)
+    return res
+
+def handle_error(response, call_name):
+    if response.status_code != 200 and response.status_code != 404:
```
Comment (jameslinnell) [inline docker/sds-api/spine-directory-service/sds/request/cpm.py:59]: remove 404

File: docker/sds-api/spine-directory-service/sds/request/cpm.py [hunk 2]
```diff
@@ -170,6 +173,68 @@ def validate_filters(self, query_parts):
         if non_empty_count < 2:
             self._raise_invalid_query_params_error()
     
+    def set_mhs_endpoint(self, ldap_results):
+        for key, ldap_result in enumerate(ldap_results):
+            service, interaction = self._extract_service_and_interaction(ldap_result['nhsMhsSvcIA'])
```
Comment (jaklinger) [inline docker/sds-api/spine-directory-service/sds/request/cpm.py:178]: I'm not sure if this is covered in other parts of the sds fhir api - but the field names coming back from ldap have cases that are all over the place, e.g. `nhsMhsSvcIA` and `nhsmhssvcia` can both be the field name from what i understand

Comment (jameslinnell) [reply docker/sds-api/spine-directory-service/sds/request/cpm.py:178]: Interesting. However this is not actual ldap it's my made up version of ldap after converting the response from CPM

File: docker/sds-api/spine-directory-service/sds/request/cpm.py [hunk 3]
```diff
@@ -170,6 +173,68 @@ def validate_filters(self, query_parts):
...
+
+                if address:
+                    ldap_results[key]['nhsMHSEndPoint'] = address
+                
+        return ldap_results
+    
+    def _get_interaction_address(self, interaction: str):
+        forward_address_cache: Optional[str] = None
+        forward_address_cache = forward_address_cache or self._get_address(interaction)
+        return forward_address_cache
```
Comment (jaklinger) [inline docker/sds-api/spine-directory-service/sds/request/cpm.py:194]: the first part of the ternary can never be true, so i'd just simplify to

```suggestion
        return self._get_address(interaction)
```

**Review by jaklinger — APPROVED** [review]

**Pull request merged**